src/reflect/scala/reflect/internal/tpe/TypeComparers.scala

```diff
@@ -351,6 +351,7 @@ trait TypeComparers {
   // @assume tp1.isHigherKinded || tp2.isHigherKinded
   def isHKSubType(tp1: Type, tp2: Type, depth: Depth): Boolean = {
     def isSub(ntp1: Type, ntp2: Type) = (ntp1.withoutAnnotations, ntp2.withoutAnnotations) match {
+      case (TypeRef(_, AnyClass, _), PolyType(_, _))                        => true                     // Any is kind-polymorphic
```

Comment:
> We have the same sort of bug in lub calculations, too.

I can imagine the correct thing to do for hk lubs would be along the lines of:

  - expand kind polymorphic `Any`/`Nothing` to, e.g, `[a, b]Any` to align with the arity of the fellow lubees
  - come up with a set of type params for the result of the lub. Each type param would have a kind (arity + bounds) derived from the corresponding tparm of the list of types. The individual bounds would themselves need to be lub-ed/glb-ed somehow.
  - substitute the result type params for the original ones in the result types of the list of types
  - lub them.

Working the example above (albeit without handling higher order type parameters yet)

```
scala> val polytps = tps.map(_.normalize)
polytps: List[$r.intp.global.Type] = List([+A]List[A], [+A]Option[A])

scala> val (tparamss, results) = polytps.map(x => (x.typeParams, x.resultType)).unzip
tparamss: List[List[$r.intp.global.Symbol]] = List(List(type A), List(type A))
results: List[$r.intp.global.Type] = List(List[A], Option[A])

scala> tparamss.map(_.map(_.typeParams))
res14: List[List[List[$r.intp.global.Symbol]]] = List(List(List()), List(List()))

scala> tparamss.map(_.map(_.info.bounds.hi))
res15: List[List[$r.intp.global.Type]] = List(List(Any), List(Any))

scala> tparamss.map(_.map(_.info.bounds.lo))
res16: List[List[$r.intp.global.Type]] = List(List(Nothing), List(Nothing))

scala> val newTparams = NoSymbol.newSyntheticTypeParams(tparamss.head.length)
newTparams: List[$r.intp.global.TypeSymbol] = List(type T0)

scala> foreachWithIndex(newTparams)((sym, i) => {val tparams = tparamss.map(_(i)); sym.setInfo(TypeBounds(lub(tparams.map(_.info.bounds.lo)), glb(tparams.map(_.info.bounds.hi))))})

scala> val results = mapWithIndex(polytps)((tp, i) => tp.resultType.instantiateTypeParams(tp.typeParams, newTparams.map(_.tpeHK)))
results: List[$r.intp.global.Type] = List(List[T0], Option[T0])

scala> PolyType(newTparams, lub(results))
res29: $r.intp.global.PolyType = [T0]Product with Serializable
```